src/common/utils/time.hpp

```diff
@@ -29,6 +35,22 @@ namespace utils
         {
             return std::chrono::steady_clock::now();
         }
+
+        // Returnds the current timestamp counter value. RDTSC on x86/x64, or just time since epoch for ARM
+        /// TODO: find better solution for ARM and Figure out better CPU base frequency heuristics
+        virtual uint64_t timestamp_counter()
+        {
+#if defined(_MSC_VER)
+#if defined(_M_X64) || defined(_M_AMD64) || defined(_M_IX86)
+            return __rdtsc(); // 64-bit with MSVC intrinsic
+#endif
+
+#elif defined(__x86_64__) || defined(__i386__) || defined(__amd64__) // If we are using clang or gcc
+            return __rdtsc(); // 64-bit with clang/gcc intrinsic
+#endif
```

Comment:
aren't those two blocks the same? 🗡️ 
```suggestion
#if defined(_M_X64) || defined(_M_AMD64) || defined(_M_IX86) || defined(__x86_64__) || defined(__i386__) || defined(__amd64__)
            return __rdtsc(); // 64-bit with MSVC intrinsic
#endif
```

Reply:
I did exactly this and committed earlier, however clang tidy got mad at it, and I can't seem to get it to run locally (it throws error in stuff like qemu and parts of ur code)

Reply:
clang tidy might complain because you might need to put the std::chrono::high... lines in the #else block, as it might get recognized as dead code otherwise